keras_nlp/models/bert.py

```diff
@@ -251,6 +266,152 @@ def get_config(self):
         return config
 
 
+TOKENIZER_DOCSTRING = """Bert tokenizer with pre-trained vocabularies.
+
+This layer will tokenize inputs as a specialized
+`keras_nlp.tokenizers.WordPieceTokenizer` which supports loading pre-trained
+vocabularies.
+
+If `pack_inputs` is True, this layer will accept a tuple of (possibly batched)
+inputs, tokenize each input, truncate them according to the `trucator` strategy,
+and pack with them appropriate `"[CLS]"` and `"[PAD]"` tokens. The output of the
+layer will be a dictionary with three keys `"token_ids"`, `"segment_ids"`, and
+`"padding_mask"`. The output can be used directly by a Bert model.
+
+If `pack_inputs` is False, this layer acts only as a raw tokenizer. Input should
```

Comment:
Worth an offline convo, but some quick thoughts. Custom packing is pretty niche, nice to be able to do it, not huge. The use cases I think are more important are more about interactively using and mix and matching things from our library.

1) Loading the bert pretrained tokenizer to try a simple transformer without a segment embedding. Or an rnn even! Think someone playing around, exploring the library. Bert vocab is pretty widespread, people might want to use it for a wide array of experimentation.
2) Writing guides and demos for Bert, and wanting to demonstrating packing to a user. It's useful to be able to show (and print), the intermediate step.

Another option is to expose a `BertTokenizer` (no packing) and a `BertPreprocessor` (with packing), but I thought the distinction would be a little confusing on first read to many. Or only expose only the raw tokenizer (no packing or constructing the dict!), but then you definitely raise the bar towards getting the whole bert modeling flow working in that case.

I think overall it would be a mistake if our only window to loading a pretrained bert vocab comes with bert packing. I would rather explore other ways to expose both the packed and unpacked versions through a different array of symbols, if we aren't too hot on this design.